### PR TITLE
Fixes #25236 - Move regenerate applicability to service class

### PR DESCRIPTION
**Pull request by sjha4**
[Pulp 3]
Move regenerate applicability to service class. This is needed only for yum repository types hence method lives in yum service class.

### CODE REVIEW

**Comment by sjha4** [comment]
[test katello]

**Comment by sjha4** [comment]
[test katello]

File: app/services/katello/pulp/repository/yum.rb [hunk 1]
```diff
@@ -55,6 +55,10 @@ def smart_proxy_download_policy
             policy
           end
         end
+
+        def regenerate_applicability
+          smart_proxy.pulp_extensions.repository.regenerate_applicability_by_ids(repo.pulp_id, true)
```
Comment (jlsherrill) [inline app/services/katello/pulp/repository/yum.rb]: I think you meam   `smart_proxy.pulp_api.extensions.repository.regenerate_applicability_by_ids`

this should fix your failing test +1 to the scenario tests :)

**Comment by sjha4** [comment]
[test katello]

**Comment by sjha4** [comment]
Thanks @jlsherrill !

**Pull request merged**